From: Yifeng <wuyifeng@nd.com.cn>
Date: Thur, 8 Dec 2022 07:40:00 +0000
Subject: Fix the issue that the home page settings do not work

Fix version: 1.2.1
why:
The 'Set-default-page-url' and 'Modify-default-url-AP9-5158' patches change the default home url to
'https://www.prometheanworld.com/', and we found if 'Settings->Homepage' is toggled off and
'Settings->Homepage->Chrome's homepage' is selected, they don't seem to work, it is caused by
changing the 'UrlConstants->NTP_URL' and 'UrlConstants->NTP_NON_NATIVE_URL' values.
How:
1. Revert the 'UrlConstants->NTP_URL' and 'UrlConstants->NTP_NON_NATIVE_URL' values.
2. Set the default value of 'ChromePreferenceKeys.HOMEPAGE_PARTNER_CUSTOMIZED_DEFAULT_URI' to
'https://www.prometheanworld.com/'.
3. Set the default 'Use for new tabs' toggle to on.
4. Fix the issue that 'New Incognito tab' doesn't open the home page.
Relate to the AP9-10977 and AP9-10970 tickets.
Relate to the 'Add-option-to-use-home-page-as-NTP.patch' patch.

---
 chrome/android/java/src/org/chromium/chrome/browser/homepage/HomepageManager.java | 4 ++--
 chrome/android/java/src/org/chromium/chrome/browser/tabmodel/ChromeTabCreator.java | 2 +-
 components/embedder_support/android/java/src/org/chromium/components/embedder_support/util/UrlConstants.java | 4 ++--
 2 files changed, 5 insertions(+), 5 deletions(-)

diff --git a/chrome/android/java/src/org/chromium/chrome/browser/homepage/HomepageManager.java b/chrome/android/java/src/org/chromium/chrome/browser/homepage/HomepageManager.java
--- a/chrome/android/java/src/org/chromium/chrome/browser/homepage/HomepageManager.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/homepage/HomepageManager.java
@@ -148,7 +148,7 @@ public class HomepageManager implements HomepagePolicyManager.HomepagePolicyStat
     public static String getDefaultHomepageUri() {

         String homepagePartnerDefaultUri = SharedPreferencesManager.getInstance().readString(
-                ChromePreferenceKeys.HOMEPAGE_PARTNER_CUSTOMIZED_DEFAULT_URI, "");
+                ChromePreferenceKeys.HOMEPAGE_PARTNER_CUSTOMIZED_DEFAULT_URI, "https://www.prometheanworld.com/");
         if (!homepagePartnerDefaultUri.equals("")) return homepagePartnerDefaultUri;

         return UrlConstants.NTP_NON_NATIVE_URL;
@@ -219,7 +219,7 @@ public class HomepageManager implements HomepagePolicyManager.HomepagePolicyStat
      *
      */
     public boolean getPrefNTPIsHomepageEnabled() {
-        return mSharedPreferencesManager.readBoolean(ChromePreferenceKeys.HOMEPAGE_NTP_IS_HOMEPAGE, false);
+        return mSharedPreferencesManager.readBoolean(ChromePreferenceKeys.HOMEPAGE_NTP_IS_HOMEPAGE, true);
     }

     /**
diff --git a/chrome/android/java/src/org/chromium/chrome/browser/tabmodel/ChromeTabCreator.java b/chrome/android/java/src/org/chromium/chrome/browser/tabmodel/ChromeTabCreator.java
--- a/chrome/android/java/src/org/chromium/chrome/browser/tabmodel/ChromeTabCreator.java
+++ b/chrome/android/java/src/org/chromium/chrome/browser/tabmodel/ChromeTabCreator.java
@@ -322,7 +322,7 @@ public class ChromeTabCreator extends TabCreator {
      * @return the created tab.
      */
     public Tab launchUrl(String url, @TabLaunchType int type, Intent intent, long intentTimestamp) {
-        if (!mIncognito && url.equals(UrlConstants.NTP_URL)) {
+        if (url.equals(UrlConstants.NTP_URL)) {
             if (HomepageManager.getInstance().getPrefNTPIsHomepageEnabled()) {
                 url = HomepageManager.getInstance().getHomepageUri();
             }
diff --git a/components/embedder_support/android/java/src/org/chromium/components/embedder_support/util/UrlConstants.java b/components/embedder_support/android/java/src/org/chromium/components/embedder_support/util/UrlConstants.java
--- a/components/embedder_support/android/java/src/org/chromium/components/embedder_support/util/UrlConstants.java
+++ b/components/embedder_support/android/java/src/org/chromium/components/embedder_support/util/UrlConstants.java
@@ -44,8 +44,8 @@ public class UrlConstants {
     public static final String ABOUT_URL = "chrome://about/";

     public static final String NTP_HOST = "newtab";
-    public static final String NTP_URL = "https://www.prometheanworld.com/";
-    public static final String NTP_NON_NATIVE_URL = "https://www.prometheanworld.com/";
+    public static final String NTP_URL = "chrome-native://newtab/";
+    public static final String NTP_NON_NATIVE_URL = "chrome://newtab/";
     public static final String NTP_ABOUT_URL = "about:newtab";

     public static final String BOOKMARKS_HOST = "bookmarks";
--
2.17.1